test/metabase/driver/common/parameters/dates_test.clj

```diff
@@ -395,7 +396,28 @@
            (is (= expected-range
                   (params.dates/date-string->range s option)))))))))
 
-(deftest relative-dates-with-starting-from-zero-must-match
+(deftest date-string->range-custom-timezone-test
+  (mt/with-clock #t "2016-06-07T12:13:55Z"
+    (test.util.misc/with-fix-local-date-time-at-zone! "UTC"
+      (testing "if an timezone-id is specified, the relative datetime should relative to now at the specified timezone"
+        (is (= {:end "2016-06-07T19:00:00", :start "2016-06-07T19:00:00"}
+               (params.dates/date-string->range "thishour" {:timezone-id "Asia/Ho_Chi_Minh"})))
+
+        (testing "even if the system timezone is changed"
+          (is (= (mt/with-system-timezone-id "UTC"
+                   (params.dates/date-string->range "thishour" {:timezone-id "Asia/Ho_Chi_Minh"}))
+                 (mt/with-system-timezone-id "Asia/Tokyo"
+                   (params.dates/date-string->range "thishour" {:timezone-id "Asia/Ho_Chi_Minh"})))))
+
+        (testing "default should be at UTC"
```

Comment:
I think the test should explicitly say the system timezone, not UTC (because it just happens to be UTC in this test)
```suggestion
        (testing "default should be at the system timezone"
```